src/pkg/services/m365/api/groups.go

```diff
@@ -154,6 +155,75 @@ func (c Groups) GetByID(
 	return group, nil
 }
 
+// GetAllSites gets all the sites that belong to a group. This is
+// necessary as private and shared channels gets their on individual
+// sites. All the other channels make use of the root site.
+func (c Groups) GetAllSites(
+	ctx context.Context,
+	identifier string,
+	errs *fault.Bus,
+) ([]models.Siteable, error) {
+	root, err := c.GetRootSite(ctx, identifier)
+	if err != nil {
+		return nil, clues.Wrap(err, "getting root site")
+	}
+
+	sites := []models.Siteable{root}
+
+	channels, err := Channels(c).GetChannels(ctx, identifier)
+	if err != nil {
+		return nil, clues.Wrap(err, "getting channels")
+	}
+
+	service, err := c.Service()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, ch := range channels {
+		if ptr.Val(ch.GetMembershipType()) == models.STANDARD_CHANNELMEMBERSHIPTYPE {
+			// Standard channels use root site
+			continue
+		}
+
+		resp, err := service.
+			Client().
+			Teams().
+			ByTeamId(identifier).
+			Channels().
+			ByChannelId(ptr.Val(ch.GetId())).
+			FilesFolder().
+			Get(ctx, nil)
+		if err != nil {
+			return nil, clues.Wrap(err, "getting files folder for channel").
+				With("channel_id", ptr.Val(ch.GetId()))
+		}
+
+		// WebURL returned here is the url to the documents folder, we
+		// have to trim that out to get the actual site's webURL
+		// https://example.sharepoint.com/sites/<site-name>/Shared%20Documents/<channelName>
+		documentWebURL := ptr.Val(resp.GetWebUrl())
+
+		// TODO(meain): is this hacky? is there a better way?
```

Comment:
if it's a full weburl, can you [parse it with the url library](https://github.com/alcionai/corso/blob/main/src/pkg/services/m365/api/sites.go#L174)?